### PR TITLE
Add script for managing colo resourse ala gce.sh

### PR DESCRIPTION
**Pull request by t-nelson**
#### Problem

We have no way to manage these fancy computers we have sitting in a rack at the colo

#### Summary of Changes

Add script for managing colo resource allocation with a workflow similar to `gce.sh`.  In addition to the features provided by `gce.sh` there are also:
 - `status` - which displays the current allocations for all resoures
 - `whoami` - which displays the caller's username as mapped the their SSH public key

NOTE: `net.sh` needs run against a working code base (master is broken ATM)
NOTE2: This requires VPN access, setup instruction in progress now

### CODE REVIEW

**Comment by mvines** [comment]
We can't plug colo into the existing net/net.sh as a cloud provider?   The duplication between colo.sh and gce.sh doesn't feel great to me

**Comment by t-nelson** [comment]
The colo stuff wound up much more compatible with the cloud provider interface than we'd initially anticipated.  I implemented the obvious parts when I realized this, but there are still some large modifications in the main `gce.sh` logic that will need reworked.  I definitely agree that this is the direction we should move toward in the future

**Comment by mvines** [comment]
How large? :)

**Comment by t-nelson** [comment]
~ https://gist.github.com/t-nelson/ffbdde0c7038d7169ed7b1abe7d518e3

**Comment by mvines** [comment]
I’d really like to merge this into gce.sh, having to maintain both seems too painful.  Is that too hard or otherwise just seem like a poor use of time from your PoV?  We could break out any needed gce.sh refactoring beforehand into separate PRs

**Comment by t-nelson** [comment]
I'll throw a couple hours at it and see where I get

**Comment by t-nelson** [comment]
This is mostly ready for review.  Shellcheck is about to puke on one last outstanding issue.  OSX check for BSDisms apprecieated!

**Comment by danpaul000** [comment]
+1 with shellcheck pacified.  Tested `colo.sh create/status/delete` on OSX with no issues.

File: net/gce.sh [hunk 1]
```diff
@@ -95,6 +107,7 @@ Manage testnet instances
  config - configure the testnet and write a config file describing it
  delete - delete the testnet
  info   - display information about the currently configured testnet
+ status - display status information of all resources (colo-only)
```
Comment (mvines) [inline net/gce.sh]: nit: remove `(colo-only)` here, and supply a stub implementation of `cloud_StatusAll` for the other three providers (it can error out)

Comment (t-nelson) [reply net/gce.sh]: Sure thing!  I'm thinking we might actually want to spin this out to another `colo-admin.sh` script that handles some of the other orthogonal features the colo will require

**Comment by t-nelson** [comment]
@mvines Any qualms with https://github.com/solana-labs/solana/pull/5854/commits/90d97b8032b6df0fcfdf3f3d97666b7d3c45936e ?

**Comment by mvines** [comment]
meh, s'ok.  let's get this PR landed, it's too damn big

**Pull request merged**